avm/res/compute/disk-encryption-set/modules/nested_keyVaultPermissions.bicep

```diff
@@ -0,0 +1,69 @@
+@description('Required. A boolean to specify whether or not the used Key Vault has RBAC authentication enabled or not.')
+param rbacAuthorizationEnabled bool = true
+
+@description('Required. The resourceID of the User Assigned Identity to assign permissions to.')
+param userAssignedIdentityResourceId string
+
+@description('Optional. Resource location.')
+param location string = resourceGroup().location
+
+@description('Required. Resource ID of the KeyVault containing the key or secret.')
+param keyVaultResourceId string
+
+@description('Required. Key URL (with version) pointing to a key or secret in KeyVault.')
+param keyName string
+
+resource keyVault 'Microsoft.KeyVault/vaults@2021-10-01' existing = {
+  name: last(split(keyVaultResourceId, '/'))!
```

Comment:
May require a scope statement in case the key vault is in a different RG than the disk encryption set

Reply:
As soon as I add the scope, the deployment of RBAC complains about its scope:
`A resource's scope must match the scope of the Bicep file for it to be deployable. You must use modules to deploy resources to a different scope.` 
I will try to catch you to discuss it.

Reply:
Touche. In that case, the scope would need to be set upon referencing this template in the `disk-encryption-set/main.bicep` file.

Reply:
I think it's already set properly there:

`scope: resourceGroup(split(keyVaultResourceId, '/')[2], split(keyVaultResourceId, '/')[4])`

Reply:
True, but just to double check:

- The `main.bicep` sets the scope for this file here to possibly a different resource group (which is as intended)
- And then we set yet another scope in the below `userAssignedIdentity` reference as the MSI could (of course) be in yet another resource group.

The only thing that strikes me as a bit odd (but maybe that's just the way it is) is, that one 'existing' resource allows to set a scope (userAssignedIdentity) and one doesn't (keyVault) :D